### PR TITLE
Adds indicatorGradient to ToggleStyle

### LINKED ISSUE #44: Why is there no option to add an indicatorGradient in ToggleStyle?

### LINKED ISSUE #46: problems with backgroundGradient and backgroundColor
There are currently two bugs when using `backgroundGradient` in `ToggleStyle`:
  - `backgroundColor` in `styleBuilder` does not overwrite `backgroundGradient` in `style`
  - `backgroundGradient` in `style` is ignored when `colorBuilder` is set without assigning `backgroundColor`

### PR DESCRIPTION
**Pull request by maeddin**
closes #44
closes #46

### CODE REVIEW

**Comment by denisgl7** [comment]
What does it take to be accepted?

**Comment by maeddin** [comment]
@denisgl7 I need to do some more testing and don't know when I'll have the time. But feel free to include this version in your project already if you need the feature now.

```
animated_toggle_switch:
    git:
      url: https://github.com/SplashByte/animated_toggle_switch.git
      ref: 44-indicator-gradient
```

**Comment by denisgl7** [comment]
Everything works great! In fact, you just need to check whether the color or the gradient will be used in priority. As far as I understand - you just put a gradient on the container. Thank you very much for adding this feature!

File: lib/src/style.dart [hunk 1]
```diff
@@ -72,9 +77,11 @@ class ToggleStyle {
           ? this
           : ToggleStyle._(
               indicatorColor: other.indicatorColor ?? indicatorColor,
+              indicatorGradient: other.indicatorGradient ??
+                  (other.indicatorColor != null ? null : indicatorGradient),
               backgroundColor: other.backgroundColor ?? backgroundColor,
               backgroundGradient: other.backgroundGradient ??
-                  (other.backgroundColor == null ? null : backgroundGradient),
+                  (other.backgroundColor != null ? null : backgroundGradient),
```
Comment (maeddin) [inline lib/src/style.dart:84]: @denisgl7 This line can cause a breaking change. That's why I need to take another look at it.

**Comment by denisgl7** [comment]
[HowBuild.md](https://github.com/splashbyte/animated_toggle_switch/files/12837120/HowBuild.md)
Can you tell me why changing the theme takes so long in your component? I tried changing the animation time and it was no use. Should I create a separate issue?

**Comment by maeddin** [comment]
@denisgl7 You can create an issue, but I won't be able to fix this problem so easily. All changes in the ToggleStyle (including all changes in the theme) are currently animated with animationDuration. I would have to rewrite the current structure to do this. Would disabling these style animations work for you for now? Then I would implement something for that.

**Comment by denisgl7** [comment]
I don't feel comfortable asking you to change anything else. I'll try to fix the problem myself :)

**Comment by maeddin** [comment]
@denisgl7 I don't think you can fix this problem from outside. Create an issue and I'll take a look at it. 👍

**Pull request merged**